### PR TITLE
feat(isDataEqual): Remove is data equal from useQuery

### PR DESCRIPTION
**Pull request by judicaelandria**
This PR remove the options ```isDataEqual``` from useQuery

### CODE REVIEW

**Comment by judicaelandria** [comment]
closes https://github.com/TanStack/query/issues/4680

**Review by TkDodo — COMMENTED** [review]
I think we need to remove references to `isDataEqual` from the docs, and also add an entry to the migration guide please

**Comment by judicaelandria** [comment]
> I think we need to remove references to `isDataEqual` from the docs, and also add an entry to the migration guide please

done

File: docs/react/guides/migrating-to-react-query-5.md [hunk 1]
```diff
@@ -26,3 +26,14 @@ if you still need to remove a query, you can use `queryClient.removeQueries({que
...
+
+### The `isDataEqual` options has been removed from useQuery
+
+Previously, This function was used to indicate whether to use previous `data` (`true`) or new data (`false`) as a resolved data for the query.
+
+You can achieve the same functionality by passing a function to `structuralSharing` instead:
+
+```diff
+- isDataEqual: () => true
++ structuralSharing: false
```
Comment (TkDodo) [inline docs/react/guides/migrating-to-react-query-5.md:38]: this is not right. it should be:

```suggestion
- isDataEqual: (oldData, newData) => customCheck(oldData, newData)
+ structuralSharing: (oldData, newData) => customCheck(oldData, newData) ? oldData : replaceEqualDeep(oldData, newData)
```

where you can import `replaceEqualDeep` from `'@tanstack/react-query'`

**Review by TkDodo — APPROVED** [review]

**Pull request merged**